packages/web3-eth/src/index.js

```diff
@@ -138,8 +140,18 @@ var Eth = function Eth() {
     this.personal = new Personal(this.currentProvider);
     this.personal.defaultAccount = this.defaultAccount;
 
+    var ContractWrapper = function ContractWrapper() {
+      Contract.apply(this, arguments);
+    };
+
+    ContractWrapper.setProvider = function() {
+      Contract.setProvider.apply(this, arguments);
+    };
+
+    util.inherits(ContractWrapper, Contract);
```

Comment:
To me, I think this is the most contentious part. 

From the [current node docs](https://nodejs.org/docs/latest/api/util.html#util_util_inherits_constructor_superconstructor):

> Note: Usage of util.inherits() is discouraged. Please use the ES6 `class` and `extends` keywords to get language level inheritance support. Also note that the two styles are [semantically incompatible](https://github.com/nodejs/node/issues/4179).

So, @frozeman my question here is whether we'd rather do things the ES6 way, per those docs. Is Web3 1.0 babelified? If so, perhaps that'd be more idiomatic.

Reply:
We are currently working on the 1.0ES6 branch. When we have that we can use it. Currently we do some Babel trans compiling, but I didn’t try if it works, as all the code is es5

Reply:
Also did you try the standalone lnstantiation? According to the docs?

Reply:
Would you rather I submitted a PR against that branch? Happy to do so - I think that answers the `util.inherit` vs `class` and `extend` question as well, then?

I didn't try standalone instantiation, as I figured existing tests covered that. This PR definitely won't fix shared state between standalone instances of `Contract`, however.

If you want to remove that shared state, then I suggest we ditch the dynamic inheritance hacks altogether and go with the usual `setProvider` thing post construction and revert back to the old `contract` factory method on `web3.eth` like in the later pre-1.0 releases.

Reply:
@frozeman Just tested and this commit cherry-picks cleanly against 1.0ES6 and all tests pass - I'm happy to do that if you like and resubmit?

Reply:
Important is that standalone still works  as expected. We currently don’t have tests for that. Feel free to add some. 

Concerning the es6 branch, I will anyway merge this branch  back into 1.0ES6 so it should be fine.

Reply:
@frozeman Provided "as expected" means "as documented," then I'll be happy to add some tests for that.

Otherwise a test like the following will not pass under this branch. Please let me know if fixing this issue is a condition for acceptance.

```javascript

Contract.setProvider(provider1);
let contract1 = new Contract(...);
Contract.setProvider(provider2);
let contract2 = new Contract(...);

assert.deepEqual(provider1, contract1.currentProvider);
assert.deepEqual(provider2,contract2.currentProvider);
```

Reply:
Actually, forget I asked - I'll add that test, make it pass, and then I think we'll be in good shape. :-)